### PR TITLE
Avoid firing RemoveRedundantTableScanPredicate when there is no TableScan predicate

### PR DESCRIPTION
**Pull request by martint**
Partially mitigates https://github.com/trinodb/trino/issues/10532

### CODE REVIEW

File: core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/RemoveRedundantTableScanPredicate.java [hunk 1]
```diff
@@ -57,7 +57,7 @@
 
     private static final Pattern<FilterNode> PATTERN =
             filter().with(source().matching(
-                    tableScan().capturedAs(TABLE_SCAN)));
+                    tableScan().capturedAs(TABLE_SCAN).matching(node -> !node.getEnforcedConstraint().isAll())));
```
Comment (findepi) [inline core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/RemoveRedundantTableScanPredicate.java:60]: The rule code below doesn't make it obvious that this is a special case..
Maybe rename the rule to `RemoveRedundantPredicateAboveTableScan`

Comment (findepi) [inline core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/RemoveRedundantTableScanPredicate.java:60]: The rule code below doesn't make it obvious that this is a special case..
Maybe rename the rule to `RemoveRedundantPredicateAboveTableScan`

**Review by findepi — COMMENTED** [review]
> Partially mitigates #10532

Which connectors it will be sufficient for?
I.e. will it help in real life, or only make testing harder?

Perhaps, instead of this, normalize expression before returning it from `RemoveRedundantTableScanPredicate`?

File: core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/RemoveRedundantTableScanPredicate.java (hunk 1 above)
Comment (martint) [reply core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/RemoveRedundantTableScanPredicate.java:60]: Agreed. The name of the rule confused me when I started looking at it.

**Comment by martint** [comment]
> Which connectors it will be sufficient for? I.e. will it help in real life, or only make testing harder?

All connectors, especially when the filter contains expressions like the one above -- it's actually independent of connector implementation. It's a bug in how the `RemoveRedundantTableScanPredicate` rule is manipulating the filter predicate.

Regardless, the rule shouldn't fire when there's no work to do, which is what I'm adding here.

**Review by sopel39 — APPROVED** [review]

**Comment by martint** [comment]
Folded into https://github.com/trinodb/trino/pull/10985

**Pull request closed without merging**